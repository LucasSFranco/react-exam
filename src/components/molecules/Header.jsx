import styled from 'styled-components'

import logo from '../../assets/logo.svg'

function Header() {
  return (
    <Self className="header">
      <a
        data-test="nyt-logo"
        target="_blank"
        rel="noreferrer"
        href="https://www.nytimes.com/"
      >
        <img
          src={logo}
          alt="The New York Times"
        />
      </a>
    </Self>
  )
}

const Self = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;

  border-bottom: 1px solid var(--accent-secondary);
  padding: .75rem 0;

  a {
    width: 185px;
    height: 25px;

    display: flex;
  }
`

export default Header
