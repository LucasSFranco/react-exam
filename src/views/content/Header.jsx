import styled from 'styled-components'

import logo from '@/assets/logo.svg'

function Header() {
  return (
    <Self>
      <a target="_blank" rel="noreferrer" href="https://www.nytimes.com/">
        <img src={logo} alt="The New York Times" />
      </a>
    </Self>
  )
}

const Self = styled.header`
  display: flex;
  align-items: center;
  justify-content: center;

  padding: .75rem 0;
  border-bottom: 1px solid var(--accent-secondary);

  a {
    width: 185px;
    height: 25px;

    display: flex;
  }
`

export default Header
