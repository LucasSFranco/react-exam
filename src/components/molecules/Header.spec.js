import React from 'react'
import { render } from '@testing-library/react'

import Header from './Header'

describe('Header', () => {

  it('properly renders', async () => {

    const { container } = render(
      <Header />
    )

    expect(container.firstChild).toMatchSnapshot()
  })

})
