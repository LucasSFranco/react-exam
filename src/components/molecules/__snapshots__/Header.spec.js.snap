// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header properly renders 1`] = `
<header
  class="sc-bdnxRM ebltpS header"
>
  <a
    href="https://www.nytimes.com/"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="The New York Times"
      src="logo.svg"
    />
  </a>
</header>
`;
